src/main/java/org/sqlite/jdbc3/JDBC3DatabaseMetaData.java

```diff
@@ -1175,12 +1229,33 @@ public ResultSet getSchemas() throws SQLException {
         if (getSchemas == null) {
             getSchemas =
                     conn.prepareStatement(
-                            "select null as TABLE_SCHEM, null as TABLE_CATALOG limit 0;");
+                            "select name as TABLE_SCHEM, null as TABLE_CATALOG from pragma_database_list order by "
+                                    + "TABLE_SCHEM;");
         }
 
         return getSchemas.executeQuery();
     }
 
+    /** @see java.sql.DatabaseMetaData#getSchemas(String, String) */
+    public ResultSet getSchemas(String catalog, String schemaPattern) throws SQLException {
+        if (schemaPattern == null) {
+            return getSchemas();
+        }
+        if ("".equals(schemaPattern)) {
+            schemaPattern = "main";
+        }
```

Comment:
I don't think we should have that behaviour. The documentation says:

> schemaPattern – a schema name; must match the schema name as it is stored in the database; null means schema name should not be used to narrow down the search.

There is no mention of a special handling when passing `"""`